{
  "question_id" : 44826654,
  "title" : "Why does Spark fail with &quot;Failed to start database &#39;metastore_db&#39;&quot; when reading CSV file?",
  "body" : "<p>I am trying to build K-Means model using Spark MLlib. I have successfully trained the model. But while saving the model om the file system, i am getting some error related to hiveContext and derby database. </p>\n\n<p>NOte- The same error I am getting while trying to load a simple cvs file as a dataframe</p>\n\n<pre><code> val htx=new org.apache.spark.sql.hive.HiveContext(sc)\n htx.read.csv(\"file:/home/rieter/Test_Streaming/sparkMlib/models/trainingData\") \n</code></pre>\n\n<p>Below is the simple snippet of error</p>\n\n<pre><code>Caused by: org.apache.derby.iapi.error.StandardException: Failed to start \ndatabase 'metastore_db' with class loader \norg.apache.spark.sql.hive.client.IsolatedClientLoader$$anon$1@1631a23a, see the \nnext exception for details.\nat org.apache.derby.iapi.error.StandardException.newException(Unknown \nSource)\n\n\n Derby may have already booted the database /data/spark/bin/metastore_db.\n</code></pre>\n\n<p>I found various answers but not even a single one has proper explanation.</p>\n",
  "link" : "https://stackoverflow.com/questions/44826654/why-does-spark-fail-with-failed-to-start-database-metastore-db-when-reading",
  "owner" : {
    "user_id" : 4029265,
    "user_type" : "registered",
    "display_name" : "ishan",
    "profile_image" : "https://www.gravatar.com/avatar/1957ae8a65e9b025813c031f3b379f6f?s=128&d=identicon&r=PG",
    "link" : "https://stackoverflow.com/users/4029265/ishan",
    "reputation" : 301,
    "accept_rate" : 41
  },
  "is_answered" : false,
  "creation_date" : 1498743287,
  "last_activity_date" : 1498743567,
  "tags" : [
    "scala",
    "apache-spark",
    "apache-spark-sql"
  ],
  "score" : 0,
  "view_count" : 6,
  "answer_count" : 1
}